### ISSUE TITLE
Handle text/plain alternative part as plaintext body

### ISSUE THREAD

**Issue by cjsoftuk** [anymail/django-anymail#252]
Mailgun appears to support setting both the HTML and Plain parts separately based on https://documentation.mailgun.com/en/latest/api-sending.html#sending

Pre-Anymail, this worked flawlessly with our SMTP server.  As volume has increased, we need to move to an ESP and Mailgun is our preference.

Our application composes & sends an email by doing something like this:

```
_message = EmailMultiAlternatives(subject=..., from_email=..., reply_to=[...], to=[...])
_message.attach_alternative(
    render_to_string(
        "email/customer_order_accept.html",
        ctx
    ),
    "text/html"
)
_message.attach_alternative(
    render_to_string(
        "email/customer_order_accept.txt",
        ctx
    ),
    "text/plain"
)
_message.send()
```

* Anymail version - 8.4
* ESP (Mailgun, SendGrid, etc.) - Mailgun
* Your ANYMAIL settings (change secrets to "redacted") - Not included, issue can be reproduced regardless of settings.
* Versions of Django, requests, python - Django 2.2.17, requests 2.21.0, Python 3.7.3
* Exact error message and/or stack trace - AnymailUnsupportedFeature: Mailgun does not support alternative part with type 'text/plain'

Having looked through the code, it seems that there's handling for a single body, not a pair of bodies (one html, one plain, as I'm under the impression it's good convention to do both).

Is this is a bug?  Is this a misunderstanding on my part?  I'm not sure, but some advice on where to look next would be appreciated.

**Comment by medmunds** [comment]
Django's `EmailMultiAlternatives` by default treats the `body` parameter as the plaintext message. See the Django docs under [*Sending alternative content types*](https://docs.djangoproject.com/en/4.0/topics/email/#sending-alternative-content-types) (the first sample code block in that section). This will work correctly with Anymail. There's also a similar example in the [Anymail readme](https://github.com/anymail/django-anymail#readme) (second code block in step 3).

What you're trying to do—attach the text/plain part separately, rather than using Django's `body` parameter—is not typical usage. But it's technically not incorrect, and we generally try to have Anymail's backends behave the same as Django's SMTP backend wherever possible. Which means this is an Anymail bug (and not Mailgun-specific).

Actually, two bugs: (1) Anymail needs to handle any text/plain alternative part as a substitute for `body` (similar to how text/html alternatives are handled now). (2) Anymail needs to check `EmailMessage.content_subtype` to determine how to handle `body`.

**Comment by medmunds** [comment]
@cjsoftuk this will be fixed in the next Anymail release, but I'd still recommend updating your code to use the message's `body` attribute for the text/plain content, rather than attaching it as an alternative. 

I'm guessing most other third-party Django email backends also missed this behavior. Changing your code will make it easier if you ever need to switch to a non-Anymail backend. (Just a quick check: neither [python-sparkpost](https://github.com/SparkPost/python-sparkpost/blob/a0a5b337b3510aa7e8ee78dc06c1d44f9c7be7af/sparkpost/django/message.py#L56-L61) nor [django-sendgrid-v5](https://github.com/sklarsa/django-sendgrid-v5/blob/352c6ac610475c2fd198833ca32ea6768658c15d/sendgrid_backend/mail.py#L431-L440) currently seems to handle text/plain alternative parts.)

**Comment by medmunds** [comment]
[Fix released in Anymail v8.5]
